### PR TITLE
feat: API to fetch workspaces for homepage

### PR DESCRIPTION
**Pull request by abhvsn**
## Description
As per updated homepage experience we are segregating the workspaces, applications and members API. This PR adds the GET workpspace API for the user with read permission.   

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/28792

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceUnitTest.java [hunk 1]
```diff
@@ -101,8 +125,16 @@ public void cleanup() {
...
-                workspaceService.archiveById(workspace.getId()).block();
+        if (workspace != null && workspace.getDeletedAt() == null) {
+            workspace = workspaceService.archiveById(workspace.getId()).block();
+        }
+
+        if (!CollectionUtils.isNullOrEmpty(workspaceIds)) {
+            Flux.fromIterable(workspaceIds)
+                    .flatMap(workspaceId -> workspaceService.archiveById(workspaceId))
+                    .map(deletedWorkspace -> workspaceIds.remove(deletedWorkspace.getId()))
+                    .blockLast();
```
Comment (nsarupr) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceUnitTest.java:136]: Any reason for blockLast as compared to collectList()

**Review by nsarupr — APPROVED** [review]
Just a minor comment, otherwise the code looks good to me.

**Comment by abhvsn** [comment]
Closing this as changes are already merged via https://github.com/appsmithorg/appsmith/commit/921a4830bab0ab9acc88a0472739c22465e05221

**Pull request closed without merging**